compiler/src/dmd/expressionsem.d

```diff
@@ -7475,7 +7475,11 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             printf("MixinExp::semantic('%s')\n", exp.toChars());
         }
 
+        const saveInDefaultArg = sc.inDefaultArg;
+        sc.inDefaultArg = false;
+
         auto e = compileIt(exp);
+        sc.inDefaultArg = saveInDefaultArg;
```

Comment:
It's not clear to me what's going on here

Reply:
This is relevant for the following test:
```
void func(string expr = mixin("\"expr2=" ~ __MODULE__ ~ "\""))
{}
```
The mixin expression is inside a default argument, so `sc.inDefaultArg` is `true`. The code now sets `sc.inDefaultArg` temporarily to `false`, so the inside of the expression can be evaluated at compile time.
It's the same problem as for the example by @dkorpel with a template, but I only fixed it for a mixin expression and not templates.